### PR TITLE
More refactoring, remember migration for the duration of the session

### PR DESCRIPTION
**Pull request by N-Pex**

### CODE REVIEW

File: ChatSecure/Classes/Model/Yap Storage/Accounts/OTRAccount+Migration.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import Foundation
+
+public extension OTRAccount {
+
+    public func needsMigration() -> Bool {
+        guard let xmppAccount = self as? OTRXMPPAccount, !OTRServerDeprecation.hasMigrated(account: xmppAccount), let jid = xmppAccount.bareJID else { return false }
+        if OTRServerDeprecation.isDeprecated(server: jid.domain) {
+            if let vCardJid = xmppAccount.vCardTemp.jid, vCardJid.isEqual(to: jid, options: XMPPJIDCompareBare) {
+                return false // Already in the migration process
```
Comment (chrisballinger) [inline ChatSecure/Classes/Model/Yap Storage/Accounts/OTRAccount+Migration.swift]: @N-Pex Can you use `-OTRAccount.isArchived` property? https://github.com/ChatSecure/ChatSecure-iOS/blob/master/ChatSecure/Classes/Model/Yap%20Storage/Accounts/OTRXMPPAccount.m#L86

File: ChatSecure/Classes/Model/Yap Storage/Accounts/OTRAccount+Migration.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+//
+//  Created by N-Pex on 2017-05-11.
+//  Copyright © 2017 Chris Ballinger. All rights reserved.
+//
+
+import Foundation
+
+public extension OTRAccount {
+
+    public func needsMigration() -> Bool {
```
Comment (chrisballinger) [inline ChatSecure/Classes/Model/Yap Storage/Accounts/OTRAccount+Migration.swift]: Could this be:

```swift
public var needsMigration: Bool {
    return blah
}
```

File: ChatSecure/Classes/Model/Yap Storage/Accounts/OTRAccount+Migration.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+//  OTRAccount+Migration.swift
+//  ChatSecure
+//
+//  Created by N-Pex on 2017-05-11.
+//  Copyright © 2017 Chris Ballinger. All rights reserved.
+//
+
+import Foundation
+
+public extension OTRAccount {
```
Comment (chrisballinger) [inline ChatSecure/Classes/Model/Yap Storage/Accounts/OTRAccount+Migration.swift]: Maybe extension on `OTRXMPPAccount`?

File: ChatSecure/Classes/Model/Yap Storage/Accounts/OTRAccount+Migration.swift [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        if OTRServerDeprecation.isDeprecated(server: jid.domain) {
+            if let vCardJid = xmppAccount.vCardTemp.jid, vCardJid.isEqual(to: jid, options: XMPPJIDCompareBare) {
+                return false // Already in the migration process
+            }
+            return true
+        }
+        return false
+    }
+    
+    public func setHasMigrated(_ migrated:Bool) {
```
Comment (chrisballinger) [inline ChatSecure/Classes/Model/Yap Storage/Accounts/OTRAccount+Migration.swift]: Document this as non thread safe

**Comment by N-Pex** [comment]
Yeah, no time at the moment, had to go for the most basic approach here. But you are right, using isArchived instead makes this PR almost unnecessary.

**Comment by chrisballinger** [comment]
Alright I'll wait to merge this upstream and then we can reconcile the Zom fork after the dust settles.

**Comment by N-Pex** [comment]
Closing this, creating a new PR with changes.

**Pull request closed without merging**